src/EventHub/EventHub/Cmdlets/NetworkRuleSet/SetAzureEventHubNetworkrule.cs

```diff
@@ -0,0 +1,115 @@
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Name { get; set; }       
+
+        [Parameter(Mandatory = false, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "Default Action for NetwrokeuleSet")]
+        [ValidateSet("Allow", "Deny", IgnoreCase = true)]
+        [PSArgumentCompleter("Allow", "Deny")]
+        [PSDefaultValue(Value ="Deny")]
+        public string DefaultAction { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipeline = true, Position = 3, HelpMessage = "List of IPRuleSet")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNWRuleSetIpRulesAttributes> IPRules { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipeline = true, Position = 4, HelpMessage = "List of VirtualNetworkRules")]
+        [ValidateNotNullOrEmpty]
+        public List<PSNWRuleSetVirtualNetworkRulesAttributes> VirtualNtewrokRules { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetInputObjectParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "NetworkruleSet Configuration Object")]
+        [ValidateNotNullOrEmpty]
+        public PSNetworkRuleSetAttributes InputObject { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetworkRuleSetResourceIdParameterSet, ValueFromPipeline = true, Position = 2, HelpMessage = "Resource ID of Namespace")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+
+        public override void ExecuteCmdlet()
+        {
+            // Update the NetworlruleSet 
+            
+            if (ShouldProcess(target: Name, action: string.Format("Update NetworkruleSet for {0} Namespace in {1} ResourceGroup", Name, ResourceGroupName)))
+            {
+                try
+                {
+
+                    if (ParameterSetName.Equals(NetwrokruleSetPropertiesParameterSet))
+                    {
+                        PSNetworkRuleSetAttributes networkRuleSetAttributes = new PSNetworkRuleSetAttributes()
+                        {
+                            DefaultAction = DefaultAction,
+                            IpRules = IPRules,
+                            VirtualNetworkRules = VirtualNtewrokRules
+                        };
+
+                        Client.CreateOrUpdateNetworkRuleSet(ResourceGroupName, Name, networkRuleSetAttributes);
+                    }
+
+                    if (ParameterSetName.Equals(NetwrokruleSetInputObjectParameterSet))
+                    {
+                        WriteObject(Client.CreateOrUpdateNetworkRuleSet(ResourceGroupName, Name, InputObject));
```

Comment:
@v-Ajnava is there a reason that we're only writing an object in this parameter set and not the others?